src/DynamoCoreWpf/ViewModels/Search/BrowserItemViewModel.cs

```diff
@@ -430,24 +421,46 @@ private void RemoveFromItems(IEnumerable<ISearchEntryViewModel> oldItems)
 
         private void AddToItems(IEnumerable<ISearchEntryViewModel> newItems)
         {
+            bool hasClasses = Items.FirstOrDefault() is ClassesNodeCategoryViewModel;
+
             foreach (var entry in newItems)
             {
-                var first = Items.Select((x, i) => new { x.Name, Idx = i })
-                    .FirstOrDefault(
-                        x =>
-                            string.Compare(
-                                x.Name,
-                                entry.Name,
-                                StringComparison.Ordinal)
-                                >= 0);
+                var nextLargerItemIndex = -1;
+                foreach (var item in Items.Where(cat => !(cat is ClassesNodeCategoryViewModel)))
+                {
+                    if (string.Compare(item.Name, entry.Name, StringComparison.Ordinal) >= 0)
+                    {
+                        nextLargerItemIndex = Items.IndexOf(item);
+                        break;
+                    }
+                }
                 // Classes must be first in any case.
                 if (entry is ClassesNodeCategoryViewModel)
+                {
                     Items.Insert(0, entry);
-                else
-                    if (first != null)
-                        Items.Insert(first.Idx, entry);
+                    continue;
+                }
+
+                // Nodecategories(i.e. namespaces) should be before members.
+                if (entry is NodeSearchElementViewModel)
+                {
+                    if (nextLargerItemIndex >= 0)
+                    {
+                        var offset = hasClasses ? 1 : 0;
+                        Items.Insert(nextLargerItemIndex + SubCategories.Count - offset, entry);
+                    }
                     else
                         Items.Add(entry);
+                }
+                if (entry is NodeCategoryViewModel)
```

Comment:
Do you mean `else if (entry is NodeCategoryViewModel)`? Just want to ensure `entry` is not inserted twice.